py-rse-index.Rmd

```diff
@@ -22,25 +22,78 @@ source(here::here("_common.R"))
 >
 > — Terry Pratchett
 
-Computers are now as essential for research as telescopes, test tubes, and libraries,
-which means that researchers need need to know how to build, use, and share software.
+Software is now as essential to research as telescopes, test tubes, and reference libraries,
+which means that researchers need need to know how to build, check, use, and share programs.
 However,
 most introductions to programming focus on developing commercial applications,
-not on exploring problems and answering questions.
-This series of books will show you how to do the latter
+not on exploring problems whose answers aren't yet known.
+Our goal is show you how to do that,
 both on your own and as part of a team.
 
 The prequels to this book introduce three core skills:
 how to write short programs to clean and analyze data in a reproducible way,
-how to keep track of what you have done,
-and how to share your software and your results with other people.
+how to use version control to keep track of what you have done,
+and how to share your software and results with other people.
 But just as some astronomers spend their careers designing telescopes,
 some researchers focus on building software to make their colleagues' lives easier.
 People who do this are called [research software engineers][rse],
-and the aim of this book is to get you ready for this role—to help you go from writing code for your own use
+and the aim of this book is to get you ready for this role—to help you go from writing code for yourself
```

Comment:
I think we need to change this bit:

"some researchers focus on building software to make their colleagues' lives easier... people who do this are called [research software engineers][rse]"

While that description fits some people who will find this book useful, I think most people (myself included as well as Amira, Jun and Sami) don't think their primary role is writing code for other people - it's just something that is part of my job as a climate scientist now. I need the RSE skills, but I don't identify as an RSE... does that make sense?

Reply:
Good catch - fixed.